openfl/utilities/workspace.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Workspace utils module."""
+
+import logging
+import os
+import shutil
+import sys
+import time
+from pathlib import Path
+from subprocess import check_call
+from sys import executable
+
+logger = logging.getLogger(__name__)
+
+
+class ExperimentWorkspace:
+    """Experiment workspace context manager."""
+
+    def __init__(
+            self,
+            experiment_name: str,
+            data_file_path: Path,
+            is_install_requirements: bool = False
+    ) -> None:
+        """Initialize workspace context manager."""
+        self.experiment_name = experiment_name
+        self.data_file_path = data_file_path
+        self.cwd = os.getcwd()
+        self.experiment_work_dir = f'{self.cwd}/{self.experiment_name}'
+        self.is_install_requirements = is_install_requirements
+
+    def _install_requirements(self):
+        """Install experiment requirements."""
+        requirements_filename = f'{self.experiment_work_dir}/requirements.txt'
+
+        if os.path.isfile(requirements_filename):
+            attempts = 10
+            for _ in range(attempts):
```

Comment:
@aleksandr-mokrov Shouldn't we raise an exception when all attempts to install the requirements failed?